### PR TITLE
[DEV-4929] Federal Account count for agency profile page

### PR DESCRIPTION
**Pull request by emily-brents**
**Description:**
Creates federal account count endpoint for the new agency profile page. 

**Technical details:**
Return counts for both the federal accounts and the treasury accounts under the specified agency and fiscal year. Uses the same base sql as the program_activity/count endpoint

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [N/A] Matview impact assessment completed
5. [X] Frontend impact assessment completed
6. [X] Data validation completed
7. [N/A] Appropriate Operations ticket(s) created
8. [X] Jira Ticket [DEV-4929](https://federal-spending-transparency.atlassian.net/browse/DEV-4929):
    - [X] Link to this Pull-Request
    - [ ] Performance evaluation of affected (API | Script | Download)
    - [ ] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No matview changes needed, no changes to frontend, no ops tickets needed
```

### CODE REVIEW

File: usaspending_api/api_docs/markdown/endpoints.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ The currently available endpoints are listed in the following table.
 | -------- | ------- | ----------- |
 |[/api/v2/agency/<TOPTIER_AGENCY_CODE\>/](/api/v2/agency/012/)|GET| Returns agency overview information for USAspending.gov's Agency Details page for agencies that have ever awarded. |
 |[/api/v2/agency/<TOPTIER_AGENCY_CODE\>/program_activity/count/](/api/v2/agency/012/program_activity/count/)|GET| Returns the count of unique Program Activity categories for the agency in a single fiscal year |
+|[/api/v2/agency/<TOPTIER_AGENCY_CODE\>/federal_account/count/](/api/v2/agency/012/federal_account/count/)|GET| Returns the count of unique Federal Accounts and Treasury Accounts for the agency in a single fiscal year |
```
Comment (BrianZito) [inline usaspending_api/api_docs/markdown/endpoints.md:33]: Are there non-unique federal accounts? I see the distinct, but I had assumed that was there to deal with the join statements in Django.

Comment (emily-brents) [reply usaspending_api/api_docs/markdown/endpoints.md:33]: I mean it's possible for an agency to have multiple treasury accounts that link back to the same federal account, so we want only the count of unique federal accounts with no duplicates. If the wording is unclear i can change it, but that's the idea.

Comment (BrianZito) [reply usaspending_api/api_docs/markdown/endpoints.md:33]: I think I'd be concerned as a user that there are two "equal" federal accounts floating around that I don't know about.

Comment (tony-sappe) [reply usaspending_api/api_docs/markdown/endpoints.md]: What do you call "equal"?
We're counting the number of federal accounts. "One" = a federal account in the agency's File B in the given fiscal year. But a federal account isn't tied to a specific year or agency, right?

File: usaspending_api/agency/v2/urls.py [hunk 2]
```diff
@@ -10,6 +11,7 @@
         include(
             [
                 path("", AgencyOverview.as_view()),
+                path("object_class/count/", FederalAccountCount.as_view()),
```
Comment (tony-sappe) [inline usaspending_api/agency/v2/urls.py:14]: URL?

File: usaspending_api/api_contracts/contracts/v2/agency/toptier_code/federal_account/count.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# Federal Account Count [/api/v2/agency/{toptier_code}/federal_account/count/{?fiscal_year}]
+
+Returns the count of unique Federal Account and Treasury Account categories in the Agency's appropriations for a single fiscal year
```
Comment (tony-sappe) [inline usaspending_api/api_contracts/contracts/v2/agency/toptier_code/federal_account/count.md:6]:
```suggestion
Returns the count of Federal Account and Treasury Account categories in the Agency's appropriations for a single fiscal year
```

File: usaspending_api/api_contracts/contracts/v2/agency/toptier_code/federal_account/count.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            }
+    + Parameters
+        + `toptier_code`: 012 (required, number)
+            The toptier code of an agency (could be a CGAG or FREC) so only numeric character strings of length 3-4 are accepted.
+        + `fiscal_year` (optional, number)
+            The desired appropriations fiscal year. Defaults to the current FY
+
++ Response 200 (application/json)
+    + Attributes
+        + `federal_account_count` (required, number)
```
Comment (tony-sappe) [inline usaspending_api/api_contracts/contracts/v2/agency/toptier_code/federal_account/count.md:26]: @kbard  added the agency code and fiscal year to the response in his PR. Might as well include more details

File: usaspending_api/agency/tests/integration/test_agency_federal_account_count.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+from model_mommy import mommy
+from rest_framework import status
+from usaspending_api.common.helpers.fiscal_year_helpers import current_fiscal_year
+
+url = "/api/v2/agency/{code}/federal_account/count/{filter}"
+
+
+@pytest.fixture
+def agency_data():
```
Comment (tony-sappe) [inline usaspending_api/agency/tests/integration/test_agency_federal_account_count.py:11]: Would using the `agency_account_data` fixture in `conftest.py` work for these tests?

**Comment by emily-brents** [comment]
> Curious how this was tested if the URL pattern is incorrect

It was correct until merging with dev.

File: usaspending_api/agency/tests/integration/test_agency_federal_account_count.py (hunk 5 above)
Comment (emily-brents) [reply usaspending_api/agency/tests/integration/test_agency_federal_account_count.py:11]: no, these test require federal account data

Comment (BrianZito) [reply usaspending_api/agency/tests/integration/test_agency_federal_account_count.py:11]: you can always inherit one set of data into another and build off of what was there before. I haven't looked into the data enough to know if that will make things simpler or more complex in this case, but it's an option.

Comment (tony-sappe) [reply usaspending_api/agency/tests/integration/test_agency_federal_account_count.py:11]: Looks simple enough to add the `FederalAccount` mock data to conftest, but I was just curious.

**Pull request merged**